### ISSUE TITLE
Add checks to see if variable is not a WP_Error

### ISSUE THREAD

**Issue by barryceelen** [10up/distributor#175]
The post id is used in [subscriptions.php](https://github.com/10up/distributor/blob/master/includes/subscriptions.php#L108) and [ExternalConnection.php](https://github.com/10up/distributor/blob/master/includes/classes/ExternalConnection.php#L76) before checking if it is a `WP_Error`.

**Comment by dkotter** [comment]
Thanks for logging this @barryceelen. Have you seen this error happen before? Just looking briefly through the code, it seems there are a few other checks in place that should prevent us from getting `WP_Error` objects here, but there might be something I'm missing.

**Comment by barryceelen** [comment]
The errors only became visible when I working on some fixes, resulting in passing around incorrect data, not in normal usage @dkotter. Might be worthwhile adding an additional check in those spots anyway, even if just for completeness/robustness.

**Comment by dkotter** [comment]
Yep, makes sense. Let me know if you want to make a PR for this. Happy to review when that's ready.

**Comment by helen** [comment]
Punting out of 1.3 until we have a PR.
